File: 2750. Ways to Split Array Into Good Subarrays/Solution.cpp
class Solution {
public:
    int numberOfGoodSubarraySplits(vector<int>& nums) {
        int n = nums.size(), left = -1;
        long long MOD = 1000000007, count = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 1) {
                if (left != -1) {
                    count = (count == 0) ? 1LL * (i - left) : (count * (i - left)) % MOD;
                }
                left = i;
            }
        }
        if (count == 0 && left != -1) return 1;
        return (int) count;
    }
};

File: 2750. Ways to Split Array Into Good Subarrays/Solution.py
class Solution:
    def numberOfGoodSubarraySplits(self, nums: List[int]) -> int:
        n, MOD, left, count = len(nums), 1000000007, -1, 0
        for i in range(n):
            if nums[i] == 1:
                if left != -1:
                    count = i - left if count == 0 else (count * (i - left)) % MOD
                left = i
        if count == 0 and left != -1: return 1
        return count
